core/src/saros/session/internal/ActivityHandler.java

```diff
@@ -383,6 +393,8 @@ public void run() {
    */
   private TransformationResult directServerActivities(List<IActivity> activities) {
 
+    assert session.isHost() && documentServer != null;
```

Comment:
Also, you could add an assertion message to make debugging easier in case this ever were to fail.